ipaserver/install/installutils.py

```diff
@@ -1037,17 +1037,20 @@ def in_container():
        systemd-detect-virt requires the whole systemd subsystem which
        isn't a reasonable require in a container.
     """
+    data_sched = ""
     if not is_hidepid():
-        with open('/proc/1/sched', 'r') as sched:
-            data_sched = sched.readline()
-    else:
-        data_sched = []
+        try:
+            with open('/proc/1/sched', 'r') as sched:
+                data_sched = sched.readline().split()[0]
+        except FileNotFoundError:
+            if not os.system('systemd-detect-virt --container'):
```

Comment:
Don't use ``os.system``. It invokes a shell and is potentially dangerous. Use ``tasks.detect_container()`` to detect containers.